### PR TITLE
enhancement/1721 - Add labelPositions to restricted areas

### LINKED ISSUE #1721: Add "labelPositions" to restricted areas
**Background**
When defining airspaces in airport files it is possible to place one or more labels at custom locations for each airspace, using the attribute "labelPositions", which is useful for airspaces with strange geometry. (This functionality is not documented yet, see issue _Missing documentation of "labelPositions" in airport files_ #1711")

**Suggestions**
This functionality is not implemented for restricted areas, but it should be pretty straight forward to implement it in the same way as for airspaces.

**Tasks**
- [x] Implement functionality for specifying labelPositions for restricted areas
- [x] Document new functionality in documentation/airport-format.md
- [x] Also document functionality for "labelPositions" when defining airspaces in this issue instead of #1711

### PR DESCRIPTION
**Pull request by OlaRune**
Resolves #1721 

The purpose of this pull request is to allow airport developers to determine the position of the label of restricted areas.

### CODE REVIEW

File: documentation/airport-format.md [hunk 1]
```diff
@@ -295,6 +296,13 @@ Position definition of the airport airspace.  Multiple airspace areas may be def
 * **ceiling** ― The highest altitude (in [flight levels](#flight-level)) included in the airspace.
 * **airspace_class** ― The FAA class of the airspace. For non-US airports, please review [this FAA airspace classification document](https://www.faasafety.gov/gslac/ALC/course_content.aspx?cID=42&sID=505&preview=true) and find the closest match based on the way the local airspace is treated.
 * **poly** ― The coordinates of the airspace. in latitude, longitude: _see [lat, lon, elev](#latitude-longitude-elevation) for formatting_
+* **labelPositions (optional)** - For each airspace an optional `labelPositions` array may be supplied. It specifies one or more positions at which text labels are to be placed, which is useful for oddly shaped airspaces where the standard placement doesn't look right. Here is an example of defining two positions:
+```json
```
Comment (n8rzz) [inline documentation/airport-format.md:300]: be sure to pad these two code blocks with an empty line above

File: src/assets/scripts/client/airport/AirportModel.js [hunk 2]
```diff
@@ -480,7 +480,14 @@ export default class AirportModel {
                 ];
             });
 
-            obj.center = vscale(vadd(coords_max, coords_min), 0.5);
+            if (area.labelPositions) {
```
Comment (n8rzz) [inline src/assets/scripts/client/airport/AirportModel.js:483]: how about this variation, which will remove the `else`:

```js
let labelPositions = [vscale(vadd(coords_max, coords_min), 0.5)];

if (area.labelPositions) {
    labelPositions = _map(
        area.labelPositions,
        (v) => DynamicPositionModel.calculateRelativePosition(v, this._positionModel, this.magneticNorth)
    );
}

obj.labelPositions = labelPositions;
```

**Review by n8rzz — APPROVED** [review]

**Comment by OlaRune** [comment]
Thanks for the review, I'll fix your comments. Added a DO NOT MERGE label until fixed.

**Comment by OlaRune** [comment]
@n8rzz @erikquinn : Alrighty then, I've done some small changes according to the comments above so this PR should be ready to merge whenever.

**Pull request merged**